### ISSUE TITLE
Bug and Suggestion

### ISSUE THREAD

**Issue by LIslime** [xIdentified/Devotions#10]
I really appreciate the plugins you've created, and I'd like to offer some suggestions and report a bug:

BUG: I have set up three rituals, A, B, and C, for three different deities, a, b, and c, respectively. However, during testing, I found that when a player worships deity A, not only can they trigger ritual a, but they can also trigger rituals b and c, which I haven't configured for deity A.

Suggestion 1: The current ritual system seems a bit basic. It would be great to add some new features to the ritual system, such as allowing the execution of server commands at the start of a ritual through configuration. This would enhance server integration with the plugin. Additionally, introducing delay commands in ritual configuration would be beneficial. For example, I could set up a ritual that requires the player to meditate for 30 seconds after killing a zombie, with a 10-second delay. This would increase the complexity of rituals and enhance player experience.

Suggestion 2: Implement penalties for players who move excessively during meditation (e.g., curses or termination of the ritual without returning the ritual items).

Suggestion 3: Add a command to terminate rituals.

**Comment by xIdentified** [comment]
Appreciate the suggestions! Everything besides suggestion 1 [added here](https://www.spigotmc.org/resources/devotions-deities-and-blessings-⛧†.113549/updates). Will add more functionality to the ritual system later but wanted to get that stuff ready asap😃

**Comment by LIslime** [comment]
Thank you very much for your attention to the issues I raised! I translated and tested it as soon as you released the new version. I found that after being notified that it is not possible to perform a ritual for deity A that does not belong to deity A, right-clicking with an item that can trigger the ritual will prompt a cooldown. This may inconvenience players. Is it possible to cancel the cooldown after players attempt to perform the wrong ritual? Additionally, could you include "you must wait [time] before performing another ritual" in the language file for easy modification?

**Comment by LIslime** [comment]
BUG:

I found that you seem to have directly bound rituals to specific items, which may cause some bugs. I just conducted a test where I assigned ritual a1 to deity A and ritual b1 to deity B, and both rituals require apples. In this scenario, rituals a1 and b1 may conflict. When I tried to perform ritual b1 for my deity B, the system prompted me that I cannot perform ritual a1 for my deity.

**Comment by xIdentified** [comment]
I'll take a look at these later today, thought I had the cooldown msg in the lang file but looks like I made a mistake somewhere. Also will try to adjust how rituals are identified. Thanks for letting me know about everything so I can get it fixed!

**Comment by xIdentified** [comment]
I adjusted things so the cooldown only starts if a valid ritual is started or an offering is made. Also added the cooldown message back to the lang file it should be `shrines.cooldown`. 

When I wrote the plugin I thought rituals should require a unique item for each one I didn't really plan for anyone to do that. Especially since interacting with the shrines I am checking what item the player is holding. I just added a warning advising against using the same item for multiple rituals. In the future I *could* change to categories of items instead of specific items and maybe that will help.
